src/main/java/org/kainos/ea/resources/ResponsibilityController.java

```diff
@@ -20,29 +24,35 @@
 public class ResponsibilityController {
 
     private final ResponsibilityService responsibilityService;
-    private final DatabaseConnector databaseConnector;
+    private ResponsibilityDao responsibilityDao;
 
     public ResponsibilityController() {
-        databaseConnector = new DatabaseConnector();
-        responsibilityService = new ResponsibilityService(new ResponsibilityDao(), databaseConnector);
+        responsibilityService = new ResponsibilityService(new ResponsibilityDao(), new DatabaseConnector());
     }
 
     public ResponsibilityController(ResponsibilityService responsibilityService) {
-        databaseConnector = new DatabaseConnector();
         this.responsibilityService = responsibilityService;
     }
 
     @GET
     @Path("/jobs/{jobId}/responsibilities")
     @Produces(MediaType.APPLICATION_JSON)
-    public Response getResponsibilitiesForJob(@PathParam("jobId") int jobId) {
+    public Response getResponsibilitiesForJob(@PathParam("jobId") int jobId) throws SQLException {
         try {
+            if (!responsibilityService.doesJobExist(jobId)) {
```

Comment:
it looks like you're already doing this check in your service method? can you remove it from here?

Reply:
Removed.